protocol/pushnotificationclient/client.go

```diff
@@ -60,13 +60,21 @@ const pushNotificationBackoffTime int64 = 2
 // RegistrationBackoffTime is the step of the exponential backoff
 const RegistrationBackoffTime int64 = 15
 
+type ServerType int
+
+const (
+	ServerTypeDefault = iota + 1
```

Comment:
This is some black magic.

Reply:
iota is one of my favourite features of go.